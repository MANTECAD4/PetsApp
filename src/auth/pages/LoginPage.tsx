import { useNavigate } from 'react-router';

export const LoginPage = () => {
	const navigate = useNavigate();

	const onLogIn = () => {
		navigate('/', {
			replace: true,
		});
	};

	return (
		<>
			<div className="container mt-5">
				<h1>Login</h1>
				<hr />
				<button onClick={onLogIn} className="btn btn-info">
					Login
				</button>
			</div>
		</>
	);
};
