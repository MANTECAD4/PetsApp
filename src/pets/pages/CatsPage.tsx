import { PetsList } from '../components';

export const CatsPage = () => {
  return (
    <>
        <h1>The most popular cats species!</h1>
        <hr/>
        <PetsList specie = { 'felis catus' } />
    </>
  );
}
