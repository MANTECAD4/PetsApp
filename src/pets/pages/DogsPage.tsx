import { PetsList } from '../components';

export const DogsPage = () => {
  return (
    <>
        <h1>The most popular dogs species!</h1>
        <hr/>
        <PetsList specie = { 'canis lupus familiaris' } />
    </>
  );
}
