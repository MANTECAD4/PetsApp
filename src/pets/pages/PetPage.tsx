import { Navigate, useNavigate, useParams } from 'react-router';
import { getPetByNameId } from '../helpers/getPetByNameId';
import { useMemo } from 'react';

export const PetPage = () => {
	const { name } = useParams();

	const currentPet = useMemo(() => getPetByNameId(name!), [name]);

	if (!currentPet) return <Navigate to="/cats" />;

	const navigate = useNavigate();

	const onNavigateBack = () => {
		// navigate(`/${(currentPet.subspecie === 'felis catus')? 'cats': 'dogs'}`);
		navigate(-1);
	};

	const petImgUrl = `${import.meta.env.BASE_URL}assets/pets-images/${
		currentPet.id
	}.jpg`;

	return (
		<div className="row mt-3 animate__animated animate__fadeInLeft">
			<div className="col-4">
				<img className="img-thumbnail" src={petImgUrl} alt={currentPet?.name} />
			</div>
			<div className="col-8">
				<h2 className="card-title">{currentPet?.name}</h2>
				<hr />
				<ul className="list-group list-group-flush">
					<li className="list-group-item">
						{' '}
						<b>Origin: </b> {currentPet?.origin}{' '}
					</li>
					<li className="list-group-item">
						{' '}
						<b>Temperament: </b> {currentPet?.temperament.join(', ')}{' '}
					</li>
					<li className="list-group-item">
						{' '}
						<b>Size: </b> {currentPet?.size}{' '}
					</li>
					<li className="list-group-item">
						{' '}
						<b>Energy level (1 - Less / 5 - More): </b>{' '}
						{currentPet?.energyLevel}{' '}
					</li>
					<li className="list-group-item">
						{' '}
						<b>Grooming Needs: </b>{' '}
						{currentPet?.groomingNeeds ? 'Yes' : 'No'}{' '}
					</li>
					<li className="list-group-item">
						{' '}
						<b>Life Span: </b> {currentPet?.lifeSpan}{' '}
					</li>
				</ul>
				<p className="card-text mt-2">{currentPet?.description}</p>
				<div className="align-items-center">
					{/* <Link 
          className = 'btn btn-primary' 
          to='/'
        >
          Back
        </Link> */}
					<button className="btn btn-primary" onClick={onNavigateBack}>
						Back
					</button>
				</div>
			</div>
		</div>
	);
};
