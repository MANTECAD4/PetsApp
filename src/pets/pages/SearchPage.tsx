import { useLocation, useNavigate } from 'react-router';
import queryString from 'query-string';
import { useForm } from '../../hooks';
import { getPetsByName } from '../helpers/';
import { useMemo } from 'react';
import { MatchAltert, PetCard } from '../components';

export const SearchPage = () => {
	const navigate = useNavigate();
	const location = useLocation();

	const query = queryString.parse(location.search);
	const { q = '' } = query;

	const pets = useMemo(() => getPetsByName(q), [q]);

	const { formState, onInputChange, onResetForm } = useForm({
		searchText: q && typeof q === 'string' ? q : '',
	});
	const { searchText } = formState;

	const onSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		if (typeof searchText !== 'string') return;
		const cleanedInput = searchText.trim().toLowerCase();
		if (cleanedInput.length <= 1) return;
		navigate(`?q=${cleanedInput}`);
		// onResetForm();
	};

	return (
		<>
			<h1>Search</h1>
			<div className="">
				<hr />
				<div className="row">
					<div className="col-5">
						<div className="text-center">
							<h4>ˁ˚ᴥ˚ˀ</h4>
						</div>
						<hr />
						<form onSubmit={(event) => onSubmitForm(event)} className="mt-2">
							<div className="row">
								<div className="col-9">
									<input
										type="text"
										placeholder="Search a pet..."
										className="form-control"
										name="searchText"
										autoComplete="off"
										value={searchText}
										onChange={onInputChange}
									/>
								</div>
								<div className="col-2">
									<button className="btn btn-outline-primary">Search</button>
								</div>
							</div>
						</form>
					</div>
					<div className="col-7">
						<MatchAltert queryParam={q} petsLength={pets.length} />
					</div>
				</div>
			</div>
			<hr />
			<div className="row rows-cols-1 row-cols-md-3 g-3">
				{pets.map((pet) => (
					<PetCard key={pet.id} pet={pet} />
				))}
			</div>
		</>
	);
};
