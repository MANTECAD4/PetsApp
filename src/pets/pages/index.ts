export { CatsPage } from './CatsPage';
export { DogsPage } from './DogsPage';
export { HomePage } from './HomePage';
export { PetPage } from './PetPage';
export { SearchPage } from './SearchPage';
