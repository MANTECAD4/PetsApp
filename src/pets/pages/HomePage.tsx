import { CatsPage } from "./CatsPage";
import { DogsPage } from "./DogsPage";

export const HomePage = () => {
    return (
        <>
            <CatsPage/>
            <DogsPage/>
        </>
    );
}
