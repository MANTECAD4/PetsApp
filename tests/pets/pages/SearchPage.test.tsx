import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { SearchPage } from '../../../src/pets/pages/SearchPage';

describe(`Tests on "SearchPage.tsx" component`, () => {
	test(`Component must match a snapshot`, () => {
		const { container } = render(
			<MemoryRouter>
				<SearchPage />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
	test(`test on queryParameters`, () => {
		const queryParam = 'siamese';
		render(
			<MemoryRouter initialEntries={[`/search=?q=${queryParam}`]}>
				<SearchPage />
			</MemoryRouter>
		);
		const input: HTMLInputElement = screen.getByRole('textbox');
		const image: HTMLImageElement = screen.getByRole('img');
		expect(input.value).toBe(queryParam);
		expect(screen.getByText(`Results for '${queryParam}'`)).toBeTruthy();
		expect(image.src).toContain(
			'assets/pets-images/d2e7f4b9-1c3d-42c8-8c1a-2e5fbcf847a1.jpg'
		);
	});
});
