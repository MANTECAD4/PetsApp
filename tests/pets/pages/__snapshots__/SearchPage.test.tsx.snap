// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on "SearchPage.tsx" component Component must match a snapshot 1`] = `
<div>
  <h1>
    Search
  </h1>
  <div
    class=""
  >
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <div
          class="text-center"
        >
          <h4>
            ˁ˚ᴥ˚ˀ
          </h4>
        </div>
        <hr />
        <form
          class="mt-2"
        >
          <div
            class="row"
          >
            <div
              class="col-9"
            >
              <input
                autocomplete="off"
                class="form-control"
                name="searchText"
                placeholder="Search a pet..."
                type="text"
                value=""
              />
            </div>
            <div
              class="col-2"
            >
              <button
                class="btn btn-outline-primary"
              >
                Search
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        class="col-7"
      >
        <div
          class="text-center"
        >
          <h4>
            龴ↀ◡ↀ龴
          </h4>
        </div>
        <hr />
        <div
          class="alert alert-warning"
        >
          Search something ... 👀
        </div>
      </div>
    </div>
  </div>
  <hr />
  <div
    class="row rows-cols-1 row-cols-md-3 g-3"
  />
</div>
`;
